src/server/users/update.ts

```diff
@@ -17,6 +29,59 @@ export async function updateUser(id: number, rawdata: UpdateUserTypes['Detailed'
     }))
 }
 
+export async function registerNewEmail(id: number, rawdata: VerifyEmailType['Detailed']): Promise<RegisterNewEmailType> {
+    const { email } = verifyEmailValidation.detailedValidate(rawdata)
+
+    const storedUser = await prismaCall(() => prisma.user.findUniqueOrThrow({
+        where: {
+            id,
+        },
+        select: {
+            ...userFilterSelection,
+            feideAccount: {
+                select: {
+                    email: true,
+                }
+            }
+        }
+    }))
+
+    // This test may not be needed if we let users change their email later. Maybe just remove this check
+    if (storedUser.emailVerified) throw new ServerError('BAD PARAMETERS', 'Brukeren er allerede verifisert')
+
+    if (email === storedUser.feideAccount?.email) {
+        await prismaCall(() => prisma.user.update({
+            where: {
+                id,
+            },
+            data: {
+                emailVerified: (new Date()).toISOString()
+            }
+        }))
+
+        return {
+            verified: true,
+            email,
+        }
+    }
+
+    if (storedUser.feideAccount?.email !== email) {
```

Comment:
The condition in this if statement will always be true due to the previous if statement. So it's unnecessary.